src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const key in actions[i].extraData) {
+        currentState[key] = actions[i].extraData[key];
+      }
+      res.push(Object.assign({}, currentState));
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const key of actions[i].keysToRemove) {
+        delete currentState[key];
+      }
+      res.push(Object.assign({}, currentState));
+    }
+
+    if (actions[i].type === 'clear') {
+      currentState = {};
+      res.push(Object.assign({}, currentState));
```

Comment:
[CODE STYLE] - don't mutate object or arrays - it will cause unexpected results later on. You should make a copy using `Object.assign` or `spread` operator. Although you are creating a copy before pushing to the result array, consider using the spread operator to clone the `currentState` before modifying it.